lib/msf/base/sessions/pingback.rb

```diff
@@ -0,0 +1,106 @@
+# -*- coding: binary -*-
+require 'msf/base'
+require 'pry'
+
+module Msf
+module Sessions
+
+###
+#
+# This class provides the ability to receive a pingback UUID
+#
+###
+class Pingback
+
+  #
+  # This interface supports basic interaction.
+  #
+  include Msf::Session
+  include Msf::Session::Basic
+
+  #
+  # Returns the type of session.
+  #
+  def self.type
+    "pingback"
+  end
+
+  def initialize(rstream, opts={})
+    super
+    self.platform ||= ""
+    self.arch     ||= ""
+    datastore = opts[:datastore]
+  end
+
+  def self.create_session(rstream, opts={})
+    Msf::Sessions::Pingback.new(rstream, opts)
+  end
+
+  def process_autoruns(datastore)
+    uuid_read
+    cleanup
+  end
+
+  def cleanup
+    if rstream
+      # this is also a best-effort
+      rstream.close rescue nil
+      rstream = nil
+    end
+  end
+
+  def uuid_read
+    uuid_raw = rstream.get_once(16, 1)
+    if uuid_raw
+      self.uuid_string = uuid_raw.each_byte.map { |b| "%02x" % b.to_i() }.join
+      print_status("Incoming UUID = #{uuid_string}")
+
+      unless @db_active == false
+        begin
+          res = Mdm::Payload.find_by uuid: uuid_string
```

Comment:
All Mdm code should live under `lib/msf/core/db_manager`. Interaction with `Mdm::Payload` should be possible via the `Msf::DBManager::Payload` module methods.